plugins/vuetify.ts

```diff
@@ -0,0 +1,58 @@
+import type { Plugin } from 'vite'
+import type {Options } from '@vuetify/loader-shared'
+import path from 'upath'
+import { resolveVuetifyBase, normalizePath, isObject } from '@vuetify/loader-shared'
+
+export function VuetifyStylesPlugin(options: Options) {
+  let configFile: string | undefined
+  const vuetifyBase = resolveVuetifyBase()
+  const tempFiles = new Map<string, string>()
+  const isNone = options.styles === 'none'
+  const usingSassVariables = isNone ? false : isObject(options.styles)
+  return {
+    name: 'vuetify-styles',
+    enforce: 'pre',
+    configResolved (config) {
+      if (isObject(options.styles)) {
+        if (path.isAbsolute(options.styles.configFile)) {
+          configFile = path.resolve(options.styles.configFile)
+        } else {
+          configFile = path.resolve(path.join(config.root || process.cwd(), options.styles.configFile))
+        }
+      }
+    },
+    async resolveId (source, importer, { custom }) {
+      if (
+        source === 'vuetify/styles' || (
+          importer &&
+          source.endsWith('.css') &&
+          isSubdir(vuetifyBase, path.isAbsolute(source) ? source : importer)
+        )
+      ) {
+        if (options.styles === 'sass') {
+          const target = source.replace(/\.css$/, '.sass')
+          return this.resolve(target, importer, { skipSelf: true, custom })
+        }
+
+        const resolution = await this.resolve(source, importer, { skipSelf: true, custom })
+        if (!resolution)
+          return
+
+        const target = resolution.id.replace(/\.css$/, '.sass')
+        tempFiles.set(target, isNone
+          ? ''
+          : `@use "${normalizePath(configFile)}"\n@use "${resolution.id}"`
```

Comment:
I think we don't need to normalize config file

Reply:
In fact with the Vite `file://` resolver we should use:
```ts
`@use "${pathToFileURL(configFile).href}"\n@use "${pathToFileURL(resolution.id).href}"`
```

Reply:
![image](https://github.com/user-attachments/assets/f2d33434-30c0-4805-84b0-6569465de9a6)

Reply:
sass needed a leading slash on windows last time I checked: https://github.com/vuetifyjs/vuetify-loader/issues/274

Reply:
@KaelWD this branch using custom `file://` loader from https://pkg.pr.new/vite@561b940

Check https://github.com/vitejs/vite/pull/17909

Reply:
previous screenshot from my Windows laptop; workiing with and without `pathToFileURL(...).href`